source/Calamari.Aws/DeleteCloudFormationCommand.cs

```diff
@@ -1,38 +1,33 @@
-﻿using System;
-using Amazon.IdentityManagement;
-using Amazon.SecurityToken;
-using Calamari.Aws.Deployment;
+﻿using System.Threading.Tasks;
 using Calamari.Aws.Deployment.CloudFormation;
 using Calamari.Aws.Integration.CloudFormation;
+using Calamari.Aws.Util;
 using Calamari.Commands.Support;
-using Calamari.Deployment;
 using Octostache;
-using Octostache.Templates;
 
 namespace Calamari.Aws
 {
-    [Command(KnownAwsCalamariCommands.Commands.DeleteAwsCloudformation, Description = "Destroy an existing AWS CloudFormation stack")]
+    [Command(KnownAwsCalamariCommands.Commands.DeleteAwsCloudFormation, Description = "Destroy an existing AWS CloudFormation stack")]
     public class DeleteCloudFormationCommand : AwsCommand
     {
         readonly ICloudFormationService cloudFormationService;
 
         public DeleteCloudFormationCommand(
             ILog log,
             IVariables variables,
-            IAmazonSecurityTokenService amazonSecurityTokenService,
-            IAmazonIdentityManagementService amazonIdentityManagementService,
+            IAmazonClientFactory amazonClientFactory,
             ICloudFormationService cloudFormationService)
-            : base(log, variables, amazonSecurityTokenService, amazonIdentityManagementService)
+            : base(log, variables, amazonClientFactory)
         {
             this.cloudFormationService = cloudFormationService;
         }
 
-        protected override void Execute(RunningDeployment deployment)
+        protected override async Task ExecuteCore()
         {
-            var stackArn = new StackArn(deployment.Variables.Get(AwsSpecialVariables.CloudFormation.StackName));
-            var waitForCompletion = ((VariableDictionary)deployment.Variables).EvaluateTruthy(variables.Get("waitForCompletion"));
+            var stackArn = new StackArn(variables.Get(SpecialVariableNames.Aws.CloudFormation.StackName));
+            var waitForCompletion = ((VariableDictionary) variables).EvaluateTruthy(variables.Get(SpecialVariableNames.Action.WaitForCompletion));
```

Comment:
I forgot there is a `variables.GetFlag` method we can use here, you can set the default value to `True`